runtime/src/main/java/com/flipkart/flux/guice/module/ContainerModule.java

```diff
@@ -56,147 +61,182 @@
-											   MetricRegistry metricRegistry) {
-		ResourceConfig resourceConfig = new ResourceConfig();
-
-		//Register codahale metrics and publish to jmx
-		resourceConfig.register(new InstrumentedResourceMethodApplicationListener(metricRegistry));
-		JmxReporter jmxReporter = JmxReporter.forRegistry(metricRegistry).build();
-
-		//register resources
-		resourceConfig.register(stateMachineResource);
-		resourceConfig.register(statusResource);
-		resourceConfig.register(clientElbResource);
-
-		resourceConfig.register(CORSFilter.class);
-		jmxReporter.start();
-		return resourceConfig;
-	}
-
-	//may not be the right module class for this. may need to be moved later.
-	@Provides
-	@Singleton
-	ObjectMapper getObjectMapper() {
-		return new ObjectMapper();
-	}
+        Server server = new Server(threadPool);
+        ServerConnector http = new ServerConnector(server, acceptorThreads, selectorThreads);
+        http.setPort(port);
+        server.addConnector(http);
+        ServletContextHandler context = new ServletContextHandler(server, "/*");
+        ServletHolder servlet = new ServletHolder(new ServletContainer(resourceConfig));
+        context.addServlet(servlet, "/*");
+
+        final InstrumentedHandler handler = new InstrumentedHandler(metricRegistry);
+        handler.setHandler(context);
+        server.setHandler(handler);
+
+        server.setStopAtShutdown(true);
+        return server;
+    }
+
+    @Named("APIResourceConfig")
+    @Singleton
+    @Provides
+    public ResourceConfig getAPIResourceConfig(StateMachineResource stateMachineResource,
+                                               StatusResource statusResource, ClientElbResource clientElbResource,
+                                               MetricRegistry metricRegistry) {
+        ResourceConfig resourceConfig = new ResourceConfig();
+
+        //Register codahale metrics and publish to jmx
+        resourceConfig.register(new InstrumentedResourceMethodApplicationListener(metricRegistry));
+        JmxReporter jmxReporter = JmxReporter.forRegistry(metricRegistry).build();
+
+        //register resources
+        resourceConfig.register(stateMachineResource);
+        resourceConfig.register(statusResource);
+        resourceConfig.register(clientElbResource);
+
+        resourceConfig.register(CORSFilter.class);
+        resourceConfig.register(RequestLoggingFilter.class);
+        resourceConfig.register(new AuthValueFactoryProvider.Binder<>(User.class));
+        jmxReporter.start();
+        return resourceConfig;
+    }
+
+    //may not be the right module class for this. may need to be moved later.
+    @Provides
+    @Singleton
+    ObjectMapper getObjectMapper() {
+        return new ObjectMapper();
+    }
+
+    @Named("dashboardSqlSessionHandler")
+    @Provides
+    @Singleton
+    public SessionHandler sqlSessionHandler(@Named("authSession.connection.driver_class") String driver,
```

Comment:
Why are we introducing db sessions? Wouldn't cookies suffice? Is this the recommended way with AuthN library?

Reply:
Yes,for ui related login sessions we need to maintain a login session in db, which currently uses a default expiry policy. This is mentioned in the authn library